core/src/test/java/org/elasticsearch/deps/joda/SimpleJodaTests.java

```diff
@@ -314,6 +314,28 @@ public void testForInvalidDatesInEpochMillis() {
         }
     }
 
+    public void testForInvalidTimeZoneWithEpochSeconds() {
+        DateTimeZone zone = DateTimeZone.forOffsetHours(1);
+        FormatDateTimeFormatter formatter = new FormatDateTimeFormatter("epoch_seconds", new DateTimeFormatterBuilder().append(new Joda.EpochTimeParser(false)).toFormatter().withZone(zone), Locale.ROOT);
+        try {
+            formatter.parser().parseDateTime("1433144433655");
+            fail("Expected IllegalArgumentException");
```

Comment:
Just as a note, LuceneTestCase (which we inherit from) offers a nice helper to shorten these try/catch constructs, it goes something like this:
```
IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> ...do Something...);
assertThat(... some assertions about e...);
```
I see that the rest of this test also still uses try/catch for most exception tests. This is okay as it is, I just wanted to mention it.